include/seqan3/alignment/multiple/library.hpp

```diff
@@ -0,0 +1,427 @@
+            return true;
+        else             // append the score to the existing sequences and return whether it was successful
+            return iter->second.try_emplace(pos, score).second;
+    }
+
+    /*!\brief Add a score to a possibly existing entry in the library.
+     * \param seq A pair of sequence indeices.
+     * \param pos A pair of position indices.
+     * \param score The score that is added to the specified position in the specified sequence pair.
+     *              If this location does not exist it will be created with value `score`.
+     */
+    void add(coord_type seq, coord_type pos, score_type score)
+    {
+        check_seq_order(seq, pos);
+
+        // try to insert a new sequence pair
+        auto [iter, inserted] = data.try_emplace(seq, map_pos_score_type{{pos, score}});
+
+        if (!inserted)   // append to an existing sequence pair
+            iter->second[pos] += score;
+    }
+
+    /*!
+     * \brief Retrieve the iterator that contains the score of pairing two positions in the specified sequences.
+     * \param seq_pos A quadruple consisting of (seq1, seq2, pos1, pos2).
+     * \return An iterator to the score if the specified entry exists and end() otherwise.
+     */
+    iterator operator[](std::tuple<size_t, size_t, size_t, size_t> seq_pos) noexcept
+    {
+        coord_type seq = {std::get<0>(seq_pos), std::get<1>(seq_pos)};
+        coord_type pos = {std::get<2>(seq_pos), std::get<3>(seq_pos)};
+        check_seq_order(seq, pos);
+        auto seq_it = data.find(seq);
+        if (seq_it != data.end())
+        {
+            auto pos_it = seq_it->second.find(pos);
+            if (pos_it != seq_it->second.end())
+                return iterator(this, seq_it, pos_it);
+        }
+        return end();
+    }
+
+    /*!\brief Retrieve a map of all scores in the two specified sequences.
+     * \param seq A pair of sequence indices. The smaller index has to be in the first position of the pair.
+     * \return A std::optional that contains a score map if the specified sequences exist and False otherwise.
+     */
+    std::optional<map_pos_score_type> operator[](std::pair<size_t, size_t> seq) const noexcept
+    {
+        assert(seq.first < seq.second); // cannot swap the order without rewriting the resulting map
+
+        auto seq_it = data.find(seq);
+        if (seq_it != data.end())
+            return seq_it->second;
+        else
+            return {};
+    }
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the library.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to end().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr iterator begin() noexcept
+    {
+        return iterator(this);
```

Comment:
```suggestion
        return iterator{this};
```
since this is a construction right?